base_rest/models/rest_service_registration.py

```diff
@@ -107,15 +114,30 @@ def _update_auth_method_controller(self, controller_class):
         define it.
         :return:
         """
-        # If the controller class doesn't have the _default_auth, we don't
-        # have to define it on every routes.
-        if not hasattr(controller_class, "_default_auth"):
-            return
-        controller_default_auth = {"auth": controller_class._default_auth}
         for _name, method in _inspect_methods(controller_class):
             routing = getattr(method, ROUTING_DECORATOR_ATTR, None)
-            if routing is not None and not routing.get("auth"):
-                routing.update(controller_default_auth)
+            if not routing:
+                continue
+            self._apply_default_if_not_set(controller_class, routing, "auth")
+            self._apply_default_if_not_set(controller_class, routing, "csrf")
+            self._apply_default_if_not_set(controller_class, routing, "save_session")
+            self._apply_default_cors_if_not_set(controller_class, routing)
+
+    def _apply_default_if_not_set(self, controller_class, routing, attr_name):
+        default_attr_name = "_default_" + attr_name
+        if hasattr(controller_class, default_attr_name) and attr_name not in routing:
+            routing[attr_name] = getattr(controller_class, default_attr_name)
+
+    def _apply_default_cors_if_not_set(self, controller_class, routing):
+        default_attr_name = "_default_cors"
+        if hasattr(controller_class, default_attr_name) and "cors" not in routing:
+            cors = getattr(controller_class, default_attr_name)
+            routing["cors"] = cors
+            if cors:
+                # add http method 'OPTIONS' required by cors if the route is
+                # restricted to specific method
+                if routing.get("methods"):
+                    routing["methods"].append("OPTIONS")
```

Comment:
```suggestion
            if cors and "OPTIONS" not in routing.get("methods", []):
                # add http method 'OPTIONS' required by cors if the route is
                # restricted to specific method
                routing["methods"].append("OPTIONS")
```

Reply:
`methods`is not required. If not specified, all the methods are allowed for the given route...
```if cors and "OPTIONS" not in routing.get("methods", ["OPTIONS"]):```